sycl/test-e2e/Matrix/joint_matrix_bfloat16_rowmajorA_rowmajorB_impl.hpp

```diff
@@ -84,11 +96,35 @@ int main() {
   big_matrix<float, MATRIX_M, MATRIX_N> MD((float *)&D);
   big_matrix<bfloat16, MATRIX_M, MATRIX_K> MA((bfloat16 *)&A);
   big_matrix<bfloat16, MATRIX_K, MATRIX_N> MB((bfloat16 *)&B);
-  matrix_multiply(MC, MA, MB);
+  matrix_multiply<TM, TN, TK, kernel_name>(MC, MA, MB);
   matrix_multiply_ref((bfloat16 *)A, (bfloat16 *)B, (float *)D, MATRIX_M,
                       MATRIX_N, MATRIX_K);
 
   bool res = matrix_compare(MATRIX_M, MATRIX_N, (float *)C, (float *)D);
   std::cout << (res ? "passed" : "failed") << std::endl;
   return !res;
 }
+
+int main() {
+  queue q;
+  std::vector<combination> combinations =
+      q.get_device()
+          .get_info<sycl::ext::oneapi::experimental::info::device::
+                        matrix_combinations>();
+  for (unsigned int i = 0; i < combinations.size(); i++) {
+    if (combinations[i].atype == matrix_type::bf16) {
+      if (combinations[i].nsize == 0 || combinations[i].nsize == 16) {
+        gemm_row_major<16, class gemm_16>();
+        break;
+      }
+      // SG size = 32 is not supported on DG2
+#if (SG_SZ != 32)
```

Comment:
This is unnecessary, as we filter entire test in SG32 folder in .cpp file with // UNSUPPORTED: gpu-intel-dg2
```suggestion
```